custom_components/bmw_connected_drive/sensor.py

```diff
@@ -142,15 +272,60 @@ def unit_of_measurement(self) -> str:
     @property
     def device_state_attributes(self):
         """Return the state attributes of the sensor."""
-        return {
+        vehicle_all_trips = self._vehicle.state.all_trips
+        vehicle_charging_profile = self._vehicle.state.charging_profile
+        vehicle_last_destinations = self._vehicle.state.last_destinations
+        result = {
             "car": self._vehicle.name,
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
+        if self._service == ALL_TRIPS:
+            attr = getattr(vehicle_all_trips, self._attribute)
+            if self._attribute in ("average_combined_consumption", "average_electric_consumption",
```

Comment:
The new available_attributes function from your PR 251 will come in nicely here.

Reply:
yep, but not here... not all fields have the same attributes, so this is still needed here